File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/repository/tickets/ITicketRepository.cs
using Proiect_MPP.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.repository.tickets
{
    public interface ITicketRepository : Repository<Ticket, int>
    {
        Ticket? findByTouristName(string name);
        List<Ticket> getByInvoiceId(int invoiceId);
        List<Ticket> getByFlightId(int flightId);
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/repositories/tickets/ITicketRepository.java
package root.proiect_mpp.repositories.tickets;

import root.proiect_mpp.domain.Ticket;
import root.proiect_mpp.repositories.Repository;

import java.util.Collection;

public interface ITicketRepository extends Repository<Ticket, Integer> {
    Ticket findByTouristName(String touristName);
    Collection<Ticket> getByInvoiceId(int invoiceID);
    Collection<Ticket> getByFlightId(int flightID);
}
